server/src/main/java/org/opensearch/search/aggregations/bucket/range/RangeAggregator.java

```diff
@@ -310,7 +311,7 @@ public ScoreMode scoreMode() {
 
     @Override
     public LeafBucketCollector getLeafCollector(LeafReaderContext ctx, final LeafBucketCollector sub) throws IOException {
-        boolean optimized = filterRewriteOptimizationContext.tryOptimize(ctx, this::incrementBucketDocCount, false);
+        boolean optimized = filterRewriteOptimizationContext.tryOptimize(ctx, this::incrementBucketDocCount, segmentMatchAll(context, ctx));
```

Comment:
Isn't false more conservative? We're always trying to skip the segment level matchAll optimization? Should we keep it as is for this bug fix?

Reply:
Leaving as is. Instead failing when the range agg attempts to construct shard level ranges.